extensions/account-registration.md

```diff
@@ -0,0 +1,187 @@
+---
+title: `account-registration` Extension
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Ed Kellett"
+    period: "2020"
+    email: "e@kellett.im"
+  -
+    name: "Val Lorentz"
+    period: "2021"
+    email: "progval+ircv3@progval.net"
+---
+
+
+## Motivation
+
+The lack of a standardized way to register an account has been a source
+of frustration for many participants for several years.
+
+This isn't the first attempt to fix it. Unlike the withdrawn `ACC` proposal,
+this effort aims to be more appealing to implementors by being severely
+restricted in scope.
+The only register-verify flows supported are those already in use in the wild.
+The authors hope that this stripped-down solution will serve as the basis
+for future organic expansion.
+
+### Capability
+
+This specification adds the `draft/register` capability, whose presence
+signifies that the server accepts the `REGISTER` command.
+
+The capability has an optional value, a comma-separated list of key-value
+pairs; the format is intended to follow the precedent set by
+[the multiline draft][multiline].
+
+The defined keys are:
+
+ * `before-connect` - if present, indicates the server supports early
+   registration, so it MUST NOT use
+   `FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`
+ * `email-required` - if present, registrations require a valid email address
+   to process
+
+Clients MUST ignore any value assigned to these keys, and MUST ignore
+any unknown key.
+
+Software implementing this work-in-progress specification MUST NOT use
+the unprefixed `register` or `account-registration` capability names.
+Instead, implementations SHOULD use the `draft/register`
+capability name to be interoperable with other software implementing
+a compatible work-in-progress version.
+
+### Commands
+
+    REGISTER {<email> | "*"} <password>
+    
+The `REGISTER` command informs the server of a request to register
```

Comment:
For Ophion, and other environments where accounts exist, but nicknames are not owned (like Undernet for example), this is somewhat awkward.

It would be nice if we could have it be:

    REGISTER {<email> | "*"} [accountname] <password>

This would work for Ophion (and presumably cases like Undernet too).

Reply:
Done by @DanielOaks through a PR to my repo.